### PR TITLE
ANALPA-3332: Selaimeen tallennetut default-parametrit käyttöliittymään

### PR DESCRIPTION
**Pull request by victorlindh**
## Jira issue
[ANALPA-3332](https://extranet.vayla.fi/jira/browse/ANALPA-3332)

## Description
Feture to continue from "where you left of" by saving field values to localstorage. Also added possibility to save & load "parameter templates", and view difference between current parameters and the selected parameter template.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🏭 Code Refactor
- [ ] 🏋🏼‍♀️ Performance Improvements
- [ ] ✅ Test
- [ ] 🦾 CI/CD

## Added tests?

- [ ] 👍 yes
- [ ] 👻 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [x] 👨🏻‍💻 documented in code: docstrings and typing
- [ ] 🙅 no documentation needed

## Checklist

- [x] The code change is tested and works locally
- [ ] Local tests pass
- [x] I have followed good coding conventions, eg [these](https://peps.python.org/pep-0008/)
- [x] There is no commented out code in this PR
- [ ] The code has been formatted using **isort .**, **pre-commit run --all-files**
- [ ] Tests have been run and they pass
- [ ] New or updated dependencies have been added to `requirements.txt` and/or `requirements_dev.txt`

### CODE REVIEW

File: src/components/UserInputComponent/DefaultParametersComponent.js [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+  const [templateName, setTemplateName] = useState("");
+  const [selectedLoadJSONString, setSelectedLoadJSONString] = useState("");
+  const [diff, setDiff] = useState([]);
+  const { setSelectedWayareaLoaded } = useContext(SelectedWayareaLoadedContext);
+  const { setSelectedRoutelineLoaded } = useContext(
+    SelectedRoutelineLoadedContext
+  );
+  const { setSelectedBoatLoaded } = useContext(SelectedBoatLoadedContext);
+  const defaults = {
+    name: "Oletus parametrit",
```
Comment (AWih) [inline src/components/UserInputComponent/DefaultParametersComponent.js]: oletus parametrit -> oletusparametrit

File: src/components/UserInputComponent/DefaultParametersComponent.js [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+    if (selectedLoadJSONString && selectedLoadJSONString !== "") {
+      const d = getDiff(
+        formik.values,
+        JSON.parse(selectedLoadJSONString).parameters
+      );
+      const dd = d.map((_) => {
+        if (_.path) {
+          const key = _.path.join("."); //can be replaced with something better
+          const parameterValueNow = _.path.reduce(
+            (o, i) => o[i],
```
Comment (AWih) [inline src/components/UserInputComponent/DefaultParametersComponent.js]: more descriptive names for "_", "o", and "i" would help readability

File: src/components/UserInputComponent/DefaultParametersComponent.js [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+                alignItems: "flex-end",
+              }}
+            >
+              <Grid item xs={4}>
+                <Typography
+                  style={{ fontSize: 16, fontWeight: 550 }}
+                  color="textSecondary"
+                  gutterBottom
+                >
+                  Parametri kokoelmien tallentaminen
```
Comment (AWih) [inline src/components/UserInputComponent/DefaultParametersComponent.js]: parametri kokoelmien -> parametrikokoelmien (muutamassa paikassa)

File: src/utils/browserStorageHelpers.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+const lastUsedParametersKey = "lastUsedParameters";
+const savedParametersKey = "savedParameters";
+
+const assign = (obj, prop, value) => {
+  if (typeof prop === "string") {
+    prop = prop.split(".");
+  }
+  if (prop.length > 1) {
+    var e = prop.shift();
```
Comment (AWih) [inline src/utils/browserStorageHelpers.js]: use `const` (or `let`) instead of `var`

File: src/utils/browserStorageHelpers.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    );
+  } else obj[prop[0]] = value;
+};
+
+const del = (obj, prop) => {
+  if (typeof prop === "string") {
+    prop = prop.split(".");
+  }
+  if (prop.length > 1) {
+    var e = prop.shift();
```
Comment (AWih) [inline src/utils/browserStorageHelpers.js]: same here, use const or let

**Review by jkeponen — APPROVED** [review]

**Review by AWih-cgi — APPROVED** [review]

**Pull request merged**